rllib/algorithms/ppo/torch/ppo_torch_policy_rlm.py

```diff
@@ -0,0 +1,302 @@
+    PPOTorchRLModule,
+    PPOModuleConfig,
+    FCConfig,
+)
+
+torch, nn = try_import_torch()
+
+logger = logging.getLogger(__name__)
+
+
+class PPOTorchPolicyWithRLModule(
+    LearningRateSchedule,
+    EntropyCoeffSchedule,
+    KLCoeffMixin,
+    TorchPolicyV2,
+):
+    """PyTorch policy class used with PPO.
+
+    This class is copied from PPOTorchPolicyV2 and is modified to support RLModules.
+    Some subtle differences:
+    - if config._enable_rl_module api is true make_rl_module should be implemented by
+    the policy the policy is assumed to be compatible with rl_modules (i.e. self.model
+    would be an RLModule)
+    - Tower stats no longer belongs to the model (i.e. RLModule) instead it belongs to
+    the policy itself.
+    - Connectors should be enabled to use this policy
+    - So far it only works for vectorized obs and action spaces (Fully connected neural
+    networks). we need model catalog to work for other obs and action spaces.
+
+    # TODO: In the future we will deprecate doing all phases of training, exploration,
+    # and inference via one policy abstraction. Instead, we will use separate
+    # abstractions for each phase. For training (i.e. gradient updates, given the
+    # sample that have been collected) we will use RLTrainer which will own one or
+    # possibly many RLModules, and RLOptimizer. For exploration, we will use RLSampler
+    # which will own RLModule, and RLTrajectoryProcessor. The exploration and inference
+    # phase details are TBD but the whole point is to make rllib extremely modular.
+    """
+
+    def __init__(self, observation_space, action_space, config):
+        config = dict(ray.rllib.algorithms.ppo.ppo.PPOConfig().to_dict(), **config)
+        # TODO: Move into Policy API, if needed at all here. Why not move this into
+        #  `PPOConfig`?.
+        validate_config(config)
+
+        TorchPolicyV2.__init__(
+            self,
+            observation_space,
+            action_space,
+            config,
+            max_seq_len=config["model"]["max_seq_len"],
+        )
+
+        LearningRateSchedule.__init__(self, config["lr"], config["lr_schedule"])
+        EntropyCoeffSchedule.__init__(
+            self, config["entropy_coeff"], config["entropy_coeff_schedule"]
+        )
+        KLCoeffMixin.__init__(self, config)
+
+        # TODO: Don't require users to call this manually.
+        self._initialize_loss_from_dummy_batch()
+
+    @override(Policy)
+    def make_rl_module(self):
+        """Returns the RLModule to use for this policy.
+
+        This implementation will be replaced with model catalog calls in the future.
+        For now we basically create the barebones of a fully connected network to match
+        the behavior of what is used to be in the old policy. This is a temporary
+        solution to get RLModules working with PPO.
+
+        """
+
+        activation = self.config["model"]["fcnet_activation"]
```

Comment:
TODO, these logics should move to the new ModelCatalog, when it's available, right?